Libplanet/Blockchain/BlockChain.cs

```diff
@@ -966,29 +980,20 @@ public long GetNextTxNonce(Address address)
             {
                 _rwlock.ExitUpgradeableReadLock();
             }
-
-            if (renderBlocks && ActionRenderers.Any())
-            {
-                if (renderActions)
-                {
-                    RenderBlock(evaluations, block, stateCompleters);
-                }
-
-                foreach (IActionRenderer<T> renderer in ActionRenderers)
-                {
-                    renderer.RenderBlockEnd(oldTip: prevTip ?? Genesis, newTip: block);
-                }
-            }
         }
 
         /// <summary>
         /// Render actions from block index of <paramref name="offset"/>.
         /// </summary>
         /// <param name="offset">Index of the block to start rendering from.</param>
+        /// <param name="renderer">The renderer to render actions.</param>
         /// <param name="stateCompleters">The strategy to complement incomplete block states.
         /// <see cref="StateCompleterSet{T}.Recalculate"/> by default.</param>
         /// <returns>The number of actions rendered.</returns>
-        internal int RenderBlocks(long offset, StateCompleterSet<T>? stateCompleters = null)
+        internal int RenderBlocks(
```

Comment:
How about renaming to `RenderActions()`? (it's too similar to `renderer.RenderBlock()`)